src/containers/Registration.tsx

```diff
@@ -96,110 +96,126 @@ const Registration = () => {
-          {t("validation:fields.confirmPassword")}
-        </StyledLabel>
-      </StyledFormFloating>
-      <p className="text-danger">
-        {errors.confirmPassword &&
-          errors.confirmPassword.message &&
-          t(`${errors.confirmPassword.message}`)}
-      </p>
-      <div className="d-grid gap-2">
-        <Button
-          className="mt-3 p-4 square rounded-pill"
-          type="submit"
-          variant="secondary"
-          size="lg"
-        >
-          {t("validation:fields.register")}
-        </Button>
-      </div>
-    </Form>
+    <section className="w-50 mt-4 m-auto d-grid gap-4">
+      <Form className="d-grid" onSubmit={handleSubmit(submitForm)}>
+        <StyledFormFloating className="form-floating">
+          <Form.Control
+            {...register("username")}
+            className="form-control"
+            type="text"
+            name="username"
+            placeholder={t("validation:fields.fullname")}
+            maxLength={50}
+          />
+          <StyledLabel
+            className="pt-1 text-muted"
+            htmlFor="floatingInputCustom"
+          >
+            {t("validation:fields.fullname")}
+          </StyledLabel>
+          <StyledLabel
+            counter
+            className="fs-6 mt-2 text-end"
+            htmlFor="floatingInputCustom"
+          >
+            {watchusername ? watchusername.length : 0} / 50
+          </StyledLabel>
+        </StyledFormFloating>
+        <p className="text-danger fst-italic fs-6 mt-1">
+          {errors.username &&
+            errors.username.message &&
+            t(`${errors.username.message}`)}
+        </p>
+        <StyledFormFloating className="form-floating mt-3">
+          <Form.Control
+            className="form-control"
+            {...register("email")}
+            type="text"
+            name="email"
+            placeholder={t("validation:fields.email")}
+          />
+          <StyledLabel
+            className="pt-1 text-muted"
+            htmlFor="floatingInputCustom"
+          >
+            {t("validation:fields.email")}
+          </StyledLabel>
+        </StyledFormFloating>
+        <p className="text-danger fst-italic mt-1 fs-6">
+          {errors.email && errors.email.message && t(`${errors.email.message}`)}
+        </p>
+        <StyledFormFloating className="form-floating mt-3">
+          <Form.Control
+            className="fs-6"
+            {...register("dateOfBirth")}
+            type="date"
+            name="dateOfBirth"
+          />
+          <StyledLabel
+            className="pt-1 text-muted"
+            htmlFor="floatingInputCustom"
+          >
+            {t("validation:fields.dateOfBirth")}
+          </StyledLabel>
+        </StyledFormFloating>
+        <p className="text-danger fst-italic mt-1 fs-6">
+          {errors.dateOfBirth &&
+            errors.dateOfBirth.message &&
+            t(`${errors.dateOfBirth.message}`)}
+        </p>
```

Comment:
Use ternary operators in the error display.
Add i18 to the registration shema, like the login shema.